geom/FieldFunctionTemplate.H

```diff
@@ -0,0 +1,41 @@
+#define _NAME field
+#include "function_overload.macro"
+
```

Comment:
This file should probably hav `Unary` in the name to be consistent with the binary case.

Reply:
@tclune
Maybe FieldUnaryTemplate.H? 

> Most comments are just thoughts. But I do need to understand about number of arguments for binary and unary operators.

@tclune, can you elaborate, is the "undef" value I'm passing along to all of these?